tools/Raven.Migrator/MigrationHelpers.cs

```diff
@@ -0,0 +1,209 @@
+                stream = File.Create(Path.Combine(basePath, fileName));
+            }
+
+            using (GetStream(configuration, stream, out var outStream))
+            using (var streamWriter = new StreamWriter(outStream))
+            using (var jsonTextWriter = new JsonTextWriter(streamWriter))
+            {
+                if (configuration.ConsoleExport == false)
+                    jsonTextWriter.Formatting = Formatting.Indented;
+
+                await jsonTextWriter.WriteStartObjectAsync();
+                await jsonTextWriter.WritePropertyNameAsync("Docs");
+                await jsonTextWriter.WriteStartArrayAsync();
+
+                foreach (var collection in configuration.CollectionsToMigrate)
+                {
+                    var mongoCollectionName = collection.Name;
+                    var ravenCollectionName = collection.NewName;
+                    if (string.IsNullOrWhiteSpace(ravenCollectionName))
+                        ravenCollectionName = mongoCollectionName;
+
+                    await migrateSingleCollection(mongoCollectionName, ravenCollectionName, jsonTextWriter, streamWriter);
+                }
+
+                if (migrateAttachments != null)
+                    await migrateAttachments(jsonTextWriter, streamWriter);
+
+                await jsonTextWriter.WriteEndArrayAsync();
+                await jsonTextWriter.WriteEndObjectAsync();
+            }
+        }
+
+        private static IDisposable GetStream(
+            AbstractMigrationConfiguration configuration,
+            Stream stream, out Stream outStream)
+        {
+            if (configuration.ConsoleExport)
+            {
+                outStream = stream;
+                return new DisposableAction(stream.Dispose);
+            }
+
+            var gzipStream = outStream = new GZipStream(stream, CompressionMode.Compress, leaveOpen: true);
+            return new DisposableAction(() =>
+            {
+                using (stream)
+                using (gzipStream)
+                {
+                    
+                }
+            });
+        }
+
+        public static void WriteDocument(
+            ExpandoObject document,
+            string documentId,
+            string collectionName,
+            List<string> propertiesToRemove,
+            JsonTextWriter jsonTextWriter,
+            List<Dictionary<string, object>> attachments = null)
+        {
+            var dictionary = (IDictionary<string, object>)document;
+            if (propertiesToRemove != null)
+            {
+                foreach (var toRemove in propertiesToRemove)
+                {
+                    dictionary.Remove(toRemove);
+                }
+            }
+
+            if (attachments == null || attachments.Count == 0)
+            {
+                dictionary[MetadataProperty] = new Dictionary<string, object>
```

Comment:
Wouldn't it be better to use this as a base and after that add 'if' condition to push the attachment entries?